src/main/java/api/Controller.java

```diff
@@ -28,40 +35,52 @@ public Hello hello(@RequestParam(value = "name", defaultValue = "team3") String
     /**
      * An endpoint that returns all 4934 movies
      *
-     * @return an ArrayList<Movie> of all the movies
+     * @return a List<Movie> of all the movies
      */
     @GetMapping("/all")
-    public ArrayList<Movie> all()
+    public List<Movie> all()
     {
-        return FetchFromCSV.all();
+        return ALL_MOVIES;
     }
 
     /**
      * An endpoint that returns a specific movie. If two movies share the same title,
      * the oldest is returned. To get a specific movie of a certain year, add the
-     * year paramter, i.e. /movie/titanic?year=1997
+     * year parameter, i.e. /movie/titanic?year=1997
      *
      * @param title the title of the movie
      * @return a Movie
-     * @throws MovieNotFoundException
+     * @throws MovieNotFoundException when movie isn't found
      */
     @GetMapping("/movie/{title}")
     public Movie movie(@PathVariable("title") String title,
                        @RequestParam(value = "year", defaultValue = "none") String year)
-            throws MovieNotFoundException
     {
-        ArrayList<Movie> list = FetchFromCSV.certainMovie(title);
+        ArrayList<Movie> matches = new ArrayList<>();
         Movie m = null;
-        if (!"none".equals(year))
+        String s1= "\\b(?i)("+title+")\\b"; // regex for searching the word start first in the titles
+        String s2= ".*(?i)"+title+".*"; // regex for searching the word start anywhere in the title
+        for (Movie movie : ALL_MOVIES)
         {
-            for (Movie movie: list)
+            if (movie.getTitle().matches(s1) || movie.getTitle().matches(s2))
             {
-                if (movie.getYear().equalsIgnoreCase(year)) m = movie;
+                matches.add(movie);
+            }
+            if (movie.getTitle().matches(s1))
+            {
+                m = movie;
```

Comment:
Okay so after some experimenting, I think the easiest way to fix the problem I described here (https://github.com/zedchance/oscars/pull/29#commitcomment-44684708) is to break after this line. That way it will return the oldest movie when there are duplicate exact titles (preserving previous behavior).